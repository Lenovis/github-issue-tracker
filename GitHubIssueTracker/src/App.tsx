/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */
import React from 'react';
import { StatusBar } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import styled, { ThemeProvider } from 'styled-components';
import { theme as importedTheme } from './assets/theme';
import Navigator from './routes/Navigator';
import { Provider } from 'react-redux';
import { configStore } from './state/store';

const { store } = configStore();
const App = () => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={importedTheme}>
        <Container>
          <StatusBar
            barStyle="light-content"
            backgroundColor={importedTheme.colors.backgroundColor}
          />
          <Navigator />
        </Container>
      </ThemeProvider>
    </Provider>
  );
};

export default App;

const Container = styled(SafeAreaView)`
  background-color: ${({ theme }) => theme.colors.backgroundColor};
  flex: 1;
`;
