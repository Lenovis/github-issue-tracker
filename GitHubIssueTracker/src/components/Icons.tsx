import React from 'react';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { theme } from '../assets/theme';

const defaultIconSize = 20;

export const DotCircleIcon = ({
  size = defaultIconSize,
  color = theme.colors.buttonGreen,
}) => <FontAwesome name="dot-circle-o" size={size} color={color} />;

export const CheckCircleIcon = ({
  size = defaultIconSize,
  color = theme.colors.purple,
}) => <Ionicons name="checkmark-circle-outline" size={size} color={color} />;

export const PullRequestIcon = ({
  size = defaultIconSize,
  color = theme.colors.buttonGreen,
}) => <Ionicons name="git-pull-request-outline" size={size} color={color} />;

export const MergedRequestIcon = ({
  size = defaultIconSize,
  color = theme.colors.purple,
}) => <Ionicons name="git-merge-outline" size={size} color={color} />;
