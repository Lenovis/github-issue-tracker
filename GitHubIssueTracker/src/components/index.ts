//Butons
export * from './buttons/DefaultButton';
export * from './buttons/BackButton';
export * from './buttons/NextButton';
//Texts
export * from './texts/HeaderText';
//Icons
export * from './Icons';
//IssueItems
export * from './IssuesComponents/listItem';
export * from './IssuesComponents/issuesFilterButton';
export * from './IssuesComponents/issueListView';
