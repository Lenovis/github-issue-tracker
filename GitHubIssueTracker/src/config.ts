export const config = {
  BASE_URL: 'https://api.github.com/',
};
