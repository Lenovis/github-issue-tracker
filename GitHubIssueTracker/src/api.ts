import axios from 'axios';
import { config } from './config';
import { Issue } from './types';

type Status = {
  status: boolean;
};

export default class Api {
  static get = async <T>(endpoint: string) => {
    const url = config.BASE_URL + endpoint;

    return axios.get<T>(url);
  };

  static getRepoIssues = async (owner: string, repo: string, page = 1) =>
    Api.get<Issue[] & Status>(`repos/${owner}/${repo}/issues?page=${page}`);
}
