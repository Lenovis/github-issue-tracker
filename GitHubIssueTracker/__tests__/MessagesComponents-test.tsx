import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { ToastMessage } from '../src/components';
import 'jest-styled-components';
import * as reactRedux from 'react-redux';

describe('renders correctly', () => {
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useDispatchMock.mockImplementation();
  });
  afterEach(() => {
    useDispatchMock.mockClear();
  });

  test('<ToastMessage infoText="test" show={true}/>', () => {
    const tree = renderer
      .create(<ToastMessage infoText="test" show={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<ToastMessage infoText="test" show={false}/>', () => {
    const tree = renderer
      .create(<ToastMessage infoText="test" show={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
