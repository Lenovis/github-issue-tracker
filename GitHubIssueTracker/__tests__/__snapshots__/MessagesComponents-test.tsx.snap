// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly <ToastMessage infoText="test" show={false}/> 1`] = `null`;

exports[`renders correctly <ToastMessage infoText="test" show={true}/> 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 100,
        "left": 0,
        "minHeight": 32,
        "position": "absolute",
        "right": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#000000",
          "borderRadius": 100,
          "justifyContent": "center",
          "minHeight": 32,
          "paddingBottom": 5,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
            "textAlign": "center",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
